fe/src/main/java/org/apache/doris/alter/Alter.java

```diff
@@ -200,12 +202,39 @@ public void processAlterTable(AlterTableStmt stmt) throws UserException {
                         }
                         Set<String> idxSet = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
                         idxSet.addAll(idx.getColumns());
-                            if (newColset.equals(idxSet)) {
-                                throw new AnalysisException("index for columns (" + String
-                                        .join(",", indexDef.getColumns()) + " ) already exist.");
+                        if (newColset.equals(idxSet)) {
+                            throw new AnalysisException("index for columns (" + String
+                                    .join(",", indexDef.getColumns()) + " ) already exist.");
+                        }
+                    }
+                    OlapTable olapTable = (OlapTable) table;
+                    for (String col : indexDef.getColumns()) {
+                        for (Column c : olapTable.getFullSchema()) {
```

Comment:
You can just use `olapTable.getColumn()` to get the column, no need to iterate all columns

Reply:
And the following type checking logic looks same as in `CreateTableStmt`.
So I think they can be extract to a method?

Reply:
the colum name from  create index clause  may not the same case with `olapTable.getColumn ()` so i must use case `equalsIgnoreCase` to find

Reply:
This method is case insensitive `olapTable.getColumn()`